core-bundle/contao/templates/twig/content_element/text.html.twig

```diff
@@ -23,7 +23,7 @@
     {% block text %}
         {% set text_attributes = attrs(text_attributes|default).addClass('rte') %}
         <div{% block text_attributes %}{{ text_attributes }}{% endblock %}>
-            {{ text|insert_tag|raw }}
+            {{ text|extract_styles_for_csp|insert_tag|raw }}
```

Comment:
I think it is very important that the name includes „csp“ otherwise I would never know that this filter adds csp headers. csp_rich_text? csp_inline_styles?

Reply:
I disagree. I prefer long filter names that are explicit. It's always better to be explicit than short. Short names don't buy anything :)

Reply:
Also, all the filters explain what they do, not in what context they can/should be used. They `lower`, `reverse`, `split` and `sort`. So should this filter imho.

Reply:
> Except for us, nobody cares what the filter does internally.

Why? If you as a user want to use CSP without `style-src unsafe-inline` you need to use this filter in places you want to allow basic formatting styles. So I think every user that wants to use this filter cares very deeply about what that does. Users who do not care will just use `unsafe-inline` (as they should).

I would never expect a filter called `rich_text` to weaken my CSP, this would be dangerous IMO.

I think `csp_inline_styles` would be a good name and in line with the naming of `insert_tag`. `insert_tag` handles (replaces) the insert tags it finds in the string. `csp_inline_styles` updates the CSP for all inline styles it finds in the string.

Reply:
I‘m fine with `csp_inline_styles`, too. I just think that `text|rich_text` would be a much better DX than `text|csp_inline_styles|insert_tag|raw`. Can we create filters that combine other filters? Then we could have both the `csp_inline_styles` filter and the `rich_text` filter.

Reply:
OK, so you would want a `rich_text` filter that is the combination of `|csp_inline_styles|insert_tag|raw`?

> Can we create filters that combine other filters?

I guess we could just call the runtime methods directly one after the other, but I’m not sure how to handle `is_safe` and `preserves_safety`.

I think we should name this filter here `csp_inline_styles` then, and create a new ticket for a “combined filter” called `rich_text` where we can discuss if and how we want to implement the chaining.